iroh-net/src/relay/iroh_relay.rs

```diff
@@ -0,0 +1,826 @@
+//! A full-fledged iroh-relay server.
+//!
+//! This module provides an API to create a full fledged iroh-relay server.  It is primarily
+//! used by the `iroh-relay` binary in this crate.
+
+use std::fmt;
+use std::future::Future;
+use std::net::SocketAddr;
+use std::pin::Pin;
+use std::sync::Arc;
+
+use anyhow::{anyhow, bail, Context, Result};
+use futures_lite::StreamExt;
+use http::response::Builder as ResponseBuilder;
+use http::{HeaderMap, Method, Request, Response, StatusCode};
+use hyper::body::Incoming;
+use iroh_metrics::inc;
+use tokio::net::{TcpListener, UdpSocket};
+use tokio::task::JoinSet;
+use tracing::{debug, error, info, info_span, instrument, trace, warn, Instrument};
+
+use crate::key::SecretKey;
+use crate::relay;
+use crate::relay::http::{ServerBuilder as DerpServerBuilder, TlsAcceptor};
+use crate::stun;
+use crate::util::AbortingJoinHandle;
+
+// Module defined in this file.
+use metrics::StunMetrics;
+
+const NO_CONTENT_CHALLENGE_HEADER: &str = "X-Tailscale-Challenge";
+const NO_CONTENT_RESPONSE_HEADER: &str = "X-Tailscale-Response";
```

Comment:
note for the future, we should really update these and call them `X-Iroh`

Reply:
https://github.com/n0-computer/iroh/issues/2344